### PR TITLE
Tests (pytest and visual) for PoloidalAdvectionArakawa

### PR DESCRIPTION
**Pull request by dominikbell**
- Add both `pytest` and visual tests for `advection.advection.PoloidalAdvectionArkawa` (in `advection.test_advection` and `advection.visual_test_advection` respectively)
- Make small changes to fullSimulation.py and pygyro/arakawa/discrete_brackets_polar.py
- Remove spline objects from pygyro/advection.advection.PoloidalAdvectionArakawa
- @FrederikSchnack please add documentation where indicated

### CODE REVIEW

File: pygyro/advection/test_advection.py [hunk 1]
```diff
@@ -253,15 +277,82 @@ def test_poloidalAdvection(dt, v, xc, yc):
...
+
+    f_vals = np.ndarray([npts[1], npts[0]])
+    final_f_vals = np.ndarray([npts[1], npts[0]])
+
+    deg = 3
+    omega = 1
+
+    domain = [[1, 14.5], [0, 2*np.pi]]
+    periodic = [False, True]
+    nkts = [n + 1 + deg * (int(p) - 1) for (n, p) in zip(npts, periodic)]
```
Comment (FrederikSchnack) [inline pygyro/advection/test_advection.py:306]: we don't need the spline functionality in the following

File: pygyro/advection/test_advection.py [hunk 2]
```diff
@@ -316,13 +407,81 @@ def test_poloidalAdvectionImplicit(dt, v, xc, yc):
...
```
Comment (FrederikSchnack) [inline pygyro/advection/test_advection.py:437]: see above

**Comment by FrederikSchnack** [comment]
Had some problems coming from commit `5fd30848863e351eec3b9b33e0df4e84501a65be`. I committed a proposed fix to `dominiksbranch`, which should probably be added to this PR. 
The PR itself looks good to me, except for maybe removing some more of the unneeded Spline functionality.

File: pygyro/advection/test_advection.py (hunk 1 above)
Comment (dominikbell) [reply pygyro/advection/test_advection.py:306]: yes, in the Arakawa tests of `test_advection` we can take out the spline functions

File: pygyro/advection/test_advection.py (hunk 2 above)
Comment (dominikbell) [reply pygyro/advection/test_advection.py:437]: see above

File: pygyro/advection/test_advection.py (hunk 1 above)
Comment (FrederikSchnack) [reply pygyro/advection/test_advection.py:306]: We can also do this in the future :+1:

**Review by FrederikSchnack — APPROVED** [review]

**Pull request merged**